core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/enclave_plan_instruction/enclave_plan_instruction_repository.go

```diff
@@ -0,0 +1,103 @@
+package enclave_plan_instruction
+
+import (
+	"encoding/json"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/database_accessors/enclave_db"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/instructions_plan"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+	bolt "go.etcd.io/bbolt"
+)
+
+var (
+	enclavePlanInstructionBucketName = []byte("enclave-plan-instruction-repository")
+)
+
+type EnclavePlanInstructionRepository struct {
+	enclaveDb *enclave_db.EnclaveDB
+}
+
+func GetOrCreateNewEnclavePlanInstructionRepository(enclaveDb *enclave_db.EnclaveDB) (*EnclavePlanInstructionRepository, error) {
+	if err := enclaveDb.Update(func(tx *bolt.Tx) error {
+		bucket, err := tx.CreateBucketIfNotExists(enclavePlanInstructionBucketName)
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred while creating the enclave plan instruction database bucket")
+		}
+		logrus.Debugf("Enclave plan instruction bucket: '%+v'", bucket)
```

Comment:
wondering what printing a bucket object looks like in the logs... Is that informative?